src/common/transformations/src/transformations/common_optimizations/push_constant_to_subgraph.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/push_constant_to_subgraph.hpp"
+
+#include <openvino/core/validation_util.hpp>
+#include <openvino/op/util/multi_subgraph_base.hpp>
+
+#include "itt.hpp"
+
+bool ov::pass::PushConstantToSubgraph::run_on_model(const std::shared_ptr<Model>& model) {
+    RUN_ON_FUNCTION_SCOPE(PushConstantToSubgraph);
+
+    bool result = false;
+    for (const auto& op : model->get_ordered_ops()) {
+        const auto multi_sub_graph_op = as_type_ptr<op::util::MultiSubGraphOp>(op);
+        if (!multi_sub_graph_op) {
+            continue;
+        }
+
+        std::unordered_map<size_t, std::shared_ptr<op::v0::Constant>> constants;
+        int remove_inputs_mask = 0;
+
```

Comment:
can you please add comments about what each code block is performing?

Reply:
done